yt/yt/server/master/node_tracker_server/node_tracker.cpp

```diff
@@ -1053,18 +1068,45 @@ class TNodeTracker
             }
         }
 
-        TRack* oldNodeRack = nullptr;
+        TRack* rack = nullptr;
 
         auto* node = FindNodeByAddress(options.DefaultAddress);
         auto isNodeNew = !IsObjectAlive(node);
         if (!isNodeNew) {
             KickOutPreviousNodeIncarnation(node, options.DefaultAddress);
-            oldNodeRack = node->GetRack();
+            rack = node->GetRack();
+        }
+        if (options.Rack) {
+            rack = FindRackByName(*options.Rack);
+
+            TDataCenter* dataCenter = nullptr;
```

Comment:
Seems like we do not need this variable here, you can just do `auto* dataCenter = FindDataCenterByName` below